src/api_server.cc

```diff
@@ -84,11 +84,17 @@ MetadataApiServer::MetadataApiServer(const Configuration& config,
       server_pool_()
 {
   for (int i : boost::irange(0, server_threads)) {
-    server_pool_.emplace_back(&HttpServer::run, &server_);
+    server_pool_.emplace_back([=]() { server_.run(); });
   }
 }
 
+void MetadataApiServer::Stop() {
+  server_.stop();
+  LOG(INFO) << "API server stopped";
+}
+
 MetadataApiServer::~MetadataApiServer() {
+  Stop();
```

Comment:
If we have Stop in this destructor, should we also call stop in MetadataAgent's destructor for consistency? I'm primary concerned about the inconsistency, I'm not sure what the negative effects would be.

Reply:
`MetadataAgent`'s destructor will deallocate both the API server and the reporter, which will invoke their respective destructors.

Reply:
`Stop()` is idempotent. It's just a notification under the covers, so it's ok to call it more than once. Calling it from the destructor guarantees that the server will also shut down cleanly when the object is deleted.

Reply:
SGTM